src/sprite/sprite.ts

```diff
@@ -118,6 +119,9 @@ export class Sprite3D extends Container3D {
         this._modelView, this._sprite.modelViewProjection)
       this._parentID = this.transform._worldID
       this._cameraTransformId = camera.transformId
+      const dir = Vec3.subtract(Camera.main.position.array, this.transform.position.array);
+      const projection = Vec3.scale(Camera.main.worldTransform.forward, Vec3.dot(dir, Camera.main.worldTransform.forward));
+      this._sprite.squareDistanceFromCamera = Vec3.squareMagnitude(projection);
```

Comment:
@GordonTombola Previoulsy I have used this code to calculate the distance from camera:

```
return Vec3.distance(
      this.worldTransform.position,
      Camera.main.worldTransform.position
    );
```

Could you please explain if your calculation is better?

Reply:
I used the above calculation in this example: https://codesandbox.io/s/github/jnsmalm/pixi3d-sandbox/tree/master/sprites-3d?file=/src/index.js:0-3678

Reply:
Sure, distance between two positions will work to know the distance from the centre, however, in the case of a billboard:

your camera's rotation will determine the rotation of your billboard, meaning the order will necessarily need to change based on the rotation of your camera, not just your position

Reply:
I'll get GIMP open and doodle a little diagram of the problem

Reply:
![image](https://user-images.githubusercontent.com/57897263/182126888-85d4035d-641b-4d8f-ba51-e6f62d85aaea.png)
here, black circle is a camera, the longer ray is the forward vector of the camera, the blue dots are the centre of sprite3ds

in this case: the blue dot that is closest in its distance happens to also be the on at the closest projected distance too.

However, consider you rotate your camera
![image](https://user-images.githubusercontent.com/57897263/182127143-2e8199e3-8014-49b9-8811-12bf2747d3bf.png)

Reply:
we are essentially projecting their distance based on the forward vector and where they are situated along the forward vector's length, this form of sorting is only really applicable to billboard sprites

Reply:
also only using squared magnitude as we want to avoid the square root calculation involved in the magnitude calculation if we can avoid it

Reply:
Thanks a lot for the detailed explanation, I imagined it would be something like this.